Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,132 @@
+
+    // Set our owner resource
+    pVectorGraphic->SetResource(pParentResource);
+
+    if (pathOrRawData.has_value())
+    {
+        std::string& strRawData = pathOrRawData.value();
+        std::size_t szCharPos = strRawData.find_first_not_of(" \t\n\r\f\v", 0);
+
+        char charAuto;
+
+        if (szCharPos != std::string::npos)
+            charAuto = strRawData.at(szCharPos);
+
+        if (charAuto && charAuto == '<')
+        {
+            if (!pVectorGraphic->LoadFromData(pathOrRawData.value()))
+            {
+                pVectorGraphic->Destroy();
+                throw std::invalid_argument("Unable to load raw SVG data (check for syntax errors)");
+            }
+        }
+        else
+        {
+            std::string strPath;
+            std::string strMetaPath;
+
+            if (CResourceManager::ParseResourcePathInput(pathOrRawData.value(), pParentResource, &strPath, &strMetaPath))
+            {
+                eAccessType  accessType = strPath[0] == '@' ? eAccessType::ACCESS_PRIVATE : eAccessType::ACCESS_PUBLIC;
+                CScriptFile* pFile = new CScriptFile(pParentResource->GetScriptID(), strMetaPath.c_str(), 52428800, accessType); // 52428800 is from CLuaFileDefs, maybe move that definition elsewhere so we can share it?
+
+                if (!pFile->Load(pParentResource, CScriptFile::MODE_READ))
+                {
+                    delete pFile;
+                    pVectorGraphic->Destroy();
+                    throw std::invalid_argument(SString("Unable to load SVG file [%s]", pathOrRawData.value().c_str()));
+                }
+
+                SString strXmlData;
+                pFile->Read(pFile->GetSize(), strXmlData);
+
+                // We don't need the file handler anymore
+                delete pFile;
+
+                if (strXmlData.empty() || !pVectorGraphic->LoadFromData(strXmlData))
+                {
+                    pVectorGraphic->Destroy();
+                    throw std::invalid_argument("Unable to load raw SVG data (check for XML syntax errors)");
+                }
+
+            }
+            else
+            {
+                pVectorGraphic->Destroy();
+                throw std::invalid_argument(SString("Unable to load SVG (file doesn't exist) [%s]", pathOrRawData.value().c_str()));
+            }
+        }
+    }
+
+    return pVectorGraphic;
+}
+
+CXMLNode* CLuaVectorGraphicDefs::SVGGetDocumentXML(lua_State* luaVM, CClientVectorGraphic* pVectorGraphic)
+{
+    CXMLNode* xmlDocument = pVectorGraphic->GetSVGDocumentXML();
+
+    if (!xmlDocument)
+        throw std::invalid_argument("Unable to get SVG XML document");
+
+    return xmlDocument;
+}
+
+void CLuaVectorGraphicDefs::SVGSetDocumentXML(lua_State* luaVM, CClientVectorGraphic* pVectorGraphic, CXMLNode* pXMLNode)
+{
+    if (!pVectorGraphic->LoadFromData(pXMLNode->ToString()))
+        throw std::invalid_argument("Unable to set SVG XML document");
+
+    return;
+}
```

Comment:
```suggestion
bool CLuaVectorGraphicDefs::SVGSetDocumentXML(CClientVectorGraphic* pVectorGraphic, CXMLNode* pXMLNode)
{
    return pVectorGraphic->LoadFromData(pXMLNode->ToString());
}
```

Reply:
Why has the throw been removed in your suggestion?

Reply:
I would rather prefer a boolean than a Lua error.